astra/src/test/java/com/slack/astra/metadata/snapshot/SnapshotMetadataSerializerTest.java

```diff
@@ -20,7 +20,8 @@ public void testSnapshotMetadataSerializer() throws InvalidProtocolBufferExcepti
     final String partitionId = "1";
 
     SnapshotMetadata snapshotMetadata =
-        new SnapshotMetadata(name, path, startTime, endTime, maxOffset, partitionId, LOGS_LUCENE9);
+        new SnapshotMetadata(
```

Comment:
Consider adding a test for what happens when you try to deserialize a snapshot that doesn't have size on populated. You can construct a new `Metadata.SnapshotMetadata` (protobuf) object using the builder, and leave the size off then try to deserialize that into a `SnapshotMetadata`.

The existing ZK snapshots won't have size on them, so we want to make sure we can still construct these SnapshotMetdata objects for the time it'll take for that size to populate to all snapshots.